update command is very memory hungry
and there's no good solution for that
as the memory usage is unexpected, just disable it

https://github.com/composer/composer/issues/1898

--- composer-1.0.0-0.alpha8.5.ac497fe/src/Composer/Command/UpdateCommand.php~	2014-06-10 17:13:12.000000000 +0300
+++ composer-1.0.0-0.alpha8.5.ac497fe/src/Composer/Command/UpdateCommand.php	2014-08-05 04:55:27.855176641 +0300
@@ -65,6 +65,8 @@
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        ini_set('memory_limit', -1);
+
         if ($input->getOption('no-custom-installers')) {
             $output->writeln('<warning>You are using the deprecated option "no-custom-installers". Use "no-plugins" instead.</warning>');
             $input->setOption('no-plugins', true);
